ProductDetailScreen.tsx

```diff
@@ -0,0 +1,153 @@
+import React, { useState, useEffect } from "react";
+import {View,Text,TouchableOpacity,Image,StyleSheet,ActivityIndicator,} from "react-native";
+import AsyncStorage from "@react-native-async-storage/async-storage";
+import { useRoute } from "@react-navigation/native";
+import { useFetch } from "./useFetch";
+
+const ProductDetailScreen = () => {
```

Comment:
When mounting this screen, you'd want to read async storage so you can find the current favorited status of the item. Currently, if I tap an item from your favorites list, this screen does not show the updated favorited status.